### PR TITLE
TokenManager storage options

### PR DESCRIPTION
**Pull request by aarongranick-okta**
- Adds a new option `tokenManager.storageKey`
- Adds a new built-in storage provider type: "memory"
- Allows custom storage provider

Related GH issues:
https://github.com/okta/okta-auth-js/issues/207
https://github.com/okta/okta-auth-js/issues/215
https://github.com/okta/okta-auth-js/issues/211

### CODE REVIEW

File: packages/okta-auth-js/test/spec/tokenManager.js [hunk 1]
```diff
@@ -932,7 +999,7 @@ describe('TokenManager', function() {
     function localStorageSetup() {
       return setupSync({
         tokenManager: {
-          type: 'localStorage'
```
Comment (swiftone) [inline packages/okta-auth-js/test/spec/tokenManager.js]: Wait, does this mean these tests haven't been working as intended the whole time?

File: packages/okta-auth-js/test/spec/tokenManager.js [hunk 2]
```diff
@@ -20,7 +20,8 @@ function setupSync(options) {
     redirectUri: 'https://example.com/redirect',
     tokenManager: {
       expireEarlySeconds: options.tokenManager.expireEarlySeconds || 0,
-      storage: options.tokenManager.type,
+      storage: options.tokenManager.storage,
```
Comment (swiftone) [inline packages/okta-auth-js/test/spec/tokenManager.js]: 💯  to this change - having the tests match the naming is one of those subtle things that pays off so much down the line.

**Comment by hboylan** [comment]
I just found this solution in the README for master branch and implemented it in my client before realizing the new version hadn't been released. Can we get a `npm publish`? 🙏

**Comment by aarongranick-okta** [comment]
@hboylan We are planning to publish this week. I will update with another comment when it is published.

**Comment by rluncasu** [comment]
is this published on npm ? doesn't look like it.

**Comment by aarongranick-okta** [comment]
@hboylan New version (2.10.0) has been published. Hopefully everything works as expected for you! If you see any issues, please open a new issue. Thanks for using our SDK.

**Pull request merged**